src/gui/src/chartsWidget.cpp

```diff
@@ -583,40 +637,58 @@ void ChartsWidget::setSTA(sta::dbSta* sta)
 
 void ChartsWidget::changeStartEndFilter()
 {
-  buckets_->negative.clear();
-  buckets_->positive.clear();
-  clearBarSets();
-
-  QStackedBarSeries* series
-      = static_cast<QStackedBarSeries*>(chart_->series().front());
+  if (resetting_menu_) {
+    resetting_menu_ = false;
+    return;
+  }
 
-  // The negative bar set is the one we appended first.
-  QBarSet* neg_set = series->barSets().front();
-  QBarSet* pos_set = series->barSets().back();
+  clearChart();
 
   StaPins end_points;
+  TimingPathList paths;
   const int filter_index = filters_menu_->currentIndex();
 
   if (filter_index > 0) {
     const int no_filter_index_offset = 1;
     const StartEndPathType path_type
         = static_cast<StartEndPathType>(filter_index - no_filter_index_offset);
 
-    // No need to remove unconstrained pins. They'll be filtered when fetching
-    // paths.
-    TimingPathList paths = fetchPathsBasedOnStartEnd(path_type);
-    end_points = getEndPointsFromPaths(paths);
+    paths = fetchPathsBasedOnStartEnd(path_type);
+    setLimits(paths);
   } else {
     end_points = stagui_->getEndPoints();
-    removeUnconstrainedPins(end_points);
+    removeUnconstrainedPinsAndSetLimits(end_points);
+  }
+
+  setBucketInterval();
+
+  if (!end_points.empty()) {
+    populateBuckets(&end_points, nullptr);
+  } else if (!paths.empty()) {
+    populateBuckets(nullptr, &paths);
   }
 
-  populateBuckets(end_points);
-  populateBarSets(*neg_set, *pos_set);
+  setVisualConfig();
 
   prev_filter_index_ = filter_index;
 }
 
+void ChartsWidget::setLimits(const TimingPathList& paths)
+{
+  sta::Unit* time_unit = sta_->units()->timeUnit();
+  for (const std::unique_ptr<TimingPath>& path : paths) {
+    const float slack = time_unit->staToUser(path->getSlack());
+
+    if (slack < min_slack_) {
+      min_slack_ = slack;
+    }
+
+    if (slack > max_slack_) {
+      max_slack_ = slack;
+    }
```

Comment:
You can use std::min and std::max for these:
`min_slack_ = std::min(slack, min_slack_);`
`max_slack_ = std::max(slack, max_slack_);`